File: backend/internal/models/posts/post.go
package models

import "time"

type Post struct {
	Id         int64     `json:"id"`
	User_id    int64     `json:"user_id"`
	Title      string    `json:"title"`
	Content    string    `json:"content"`
	Created_at time.Time `json:"created_at"`
}

File: frontend/static/js/post.js

import { fetchData } from './forum.js';
import { status } from './status.js';

async function creatPost(categoriesSelected) {
         let content = document.querySelector("#content")
        console.log(content.value, categoriesSelected);
        const response = await fetch("/api/post", {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                'Accept': 'application/json',
            },
            body: JSON.stringify({
                title: "title 2",
                content: content.value,
                name: categoriesSelected
            })
        })
        if (response.ok) {
            await fetchData()
            const data = await response.json();
           
            console.log("Success:", data);

        }else if (!response.ok) {
           await status(response)
        }
        console.log(response);
    
}
export {
    creatPost
}